inf-ruby.el

```diff
@@ -103,17 +103,24 @@ returns a string."
                            inf-ruby-implementations))
   :group 'inf-ruby)
 
+(defcustom inf-ruby-wrapper-command ""
+  "Launcher command pattern to format the auto-detected command.
+Useful for running the shell in another host or a container (such as Docker). Must include %s."
+  :type 'string
+  :group 'inf-ruby)
+
 (defun inf-ruby--irb-command ()
   (let ((command "irb --prompt default -r irb/completion --noreadline"))
     (when (inf-ruby--irb-needs-nomultiline-p)
       (setq command (concat command " --nomultiline")))
     command))
 
 (defun inf-ruby--irb-needs-nomultiline-p (&optional with-bundler)
-  (let* ((output (car (last (apply #'process-lines
-                                   (append
-                                    (when with-bundler '("bundle" "exec"))
-                                    '("irb" "-v"))))))
+  "Check if IRB needs the --nomultiline argument.
+WITH-BUNDLER, the command is wrapped with `bundle exec'."
+  (let* ((command (format (if (string-empty-p inf-ruby-wrapper-command) "%s" inf-ruby-wrapper-command)
+                          (concat (when with-bundler "bundle exec ") "irb -v")))
+         (output (car (last (apply #'process-lines (split-string-and-unquote command)))))
```

Comment:
I would have used `split-string-shell-command`, but that's only available starting Emacs 28.1, so I don't know if that's okay

Reply:
I think we'd rather use `process-file-shell-command` directly, similar to how `process-lines` is itself defined (but it uses `call-process`, which we'd replace). By calling shell command, we can avoid the need for splitting it.

LMK if this makes sense to you, and if it doesn't sound too hard.

Reply:
I'm not sure I understood this one, as `process-file-shell-command` doesn't seem to return the command output. I'm not sure how to, or if it's even a good idea to create a temporary buffer to capture the command output, as I'm not yet very well versed in elisp